static/space/services/chat.html

```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
+<script type=module>
+
+import Observers from '../js/Observers.js';
+import Service from '../js/Service.js';
+
+class Chat {
+  constructor() {
+    this.observers = new Observers();
+  }
+
+  setWs(ws) {
+    this.ws = ws;
+    ws.observe('chat', message => {
+      this.observers.fire('message', message);
+    })
+  }
+}
+
+const chat = new Chat();
+Service.get('ws', ws => chat.setWs(ws));
+
+class ChatClient {
+  constructor(context) {
+    this.context = context;
+  }
+
+  observe(key, cb) {
+    // TODO: should this just proxy to `chat.ws.observers.add('chat', this.context, cb)`?
```

Comment:
It definitely could be, although once `Chat` keeps a buffer of messages locally, this would be the place to replay it, which would make it possible to reload overlay.html without losing anything.